### PR TITLE
switch to grpc for deploy and eval

### PR DESCRIPTION
**Pull request by HuiyingLi**
# What does this PR do ?

Using PyTriton grpc client instead of rest API when running evaluation due to the large logits payload.
On arc_challenge with bs=1, llama3-8b
Before (rest API): 9487.118s
After (gRPC client): 502.976

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by hemildesai — COMMENTED** [review]
Great speedup, I am wondering would it be possible to remove the rest option all together from eval? There doesn't seem to be any benefit in keeping the eval with rest API as an option.

**Comment by HuiyingLi** [comment]
> Great speedup, I am wondering would it be possible to remove the rest option all together from eval? There doesn't seem to be any benefit in keeping the eval with rest API as an option.

Thanks for the suggestion! Updated the PR and removed the rest API part.

File: nemo/collections/llm/api.py [hunk 1]
```diff
@@ -326,7 +326,8 @@ def deploy(
     model_type: str = "llama",
     triton_model_name: str = "triton_model",
     triton_model_version: Optional[int] = 1,
-    triton_port: int = 8000,
+    triton_http_port: int = 8000,
+    triton_grpc_port: int = 8001,
     triton_http_address: str = "0.0.0.0",
```
Comment (athitten) [inline nemo/collections/llm/api.py:331]: Nit: Can we change `triton_port` to `triton_http_port` in the docstring as well ? 
Also can we add `triton_grpc_port` in the docstrings ?

File: nemo/collections/llm/evaluation/utils.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+def str_list2numpy(str_list: List[str]) -> np.ndarray:
+    """
+    Convert a list of strings to a numpy array of strings.
+    """
+    str_ndarray = np.array(str_list)[..., np.newaxis]
+    return np.char.encode(str_ndarray, "utf-8")
+
+
+def query_llm(
```
[the remark this answers is not in the document]
Comment (HuiyingLi) [reply nemo/collections/llm/evaluation/utils.py]: @athitten  Thank you for pointing out! Just changed to use the exisiting query_llm.

File: nemo/collections/llm/api.py (hunk 1 above)
Comment (HuiyingLi) [reply nemo/collections/llm/api.py:331]: @athitten  Thanks! Just changed the docstring accordingly.

**Review by athitten — APPROVED** [review]
LGTM! @HuiyingLi thank you very much for this PR! Also tested it on my end on MMLU, Lambada and got expected results. Very good speed up.

**Review by hemildesai — APPROVED** [review]
